src/js/patch.ts

```diff
@@ -1,1039 +1,1039 @@
-  }
- 
- // TODO - separate flag for rescaling monsters???
-};
-
-// Map of base prices.  (Tools are positive, armors are ones-complement.)
-const BASE_PRICES: {[itemId: number]: number} = {
-  // Armors
-  0x0d: 4,    // carapace shield
-  0x0e: 16,   // bronze shield
-  0x0f: 50,   // platinum shield
-  0x10: 325,  // mirrored shield
-  0x11: 1000, // ceramic shield
-  0x12: 2000, // sacred shield
-  0x13: 4000, // battle shield
-  0x15: 6,    // tanned hide
-  0x16: 20,   // leather armor
-  0x17: 75,   // bronze armor
-  0x18: 250,  // platinum armor
-  0x19: 1000, // soldier suit
-  0x1a: 4800, // ceramic suit
-  // Tools
-  0x1d: 25,   // medical herb
-  0x1e: 30,   // antidote
-  0x1f: 45,   // lysis plant
-  0x20: 40,   // fruit of lime
-  0x21: 36,   // fruit of power
-  0x22: 200,  // magic ring
-  0x23: 150,  // fruit of repun
-  0x24: 65,   // warp boots
-  0x26: 300,  // opel statue
-  // 0x31: 50, // alarm flute
-};
-
-/////////
-/////////
-/////////
-
-// const identifyKeyItemsForDifficultyBuffs = (rom: Rom) => {
-//   // // Tag key items for difficulty buffs
-//   // for (const get of rom.itemGets) {
-//   //   const item = ITEMS.get(get.itemId);
-//   //   if (!item || !item.key) continue;
-//   //   get.key = true;
-//   // }
-//   // // console.log(report);
-//   for (let i = 0; i < 0x49; i++) {
-//     // NOTE - special handling for alarm flute until we pre-patch
-//     const unique = (rom.prg[0x20ff0 + i] & 0x40) || i === 0x31;
-//     const bit = 1 << (i & 7);
-//     const addr = 0x1e110 + (i >>> 3);
-//     rom.prg[addr] = rom.prg[addr] & ~bit | (unique ? bit : 0);
-//   }
-// };
-
-// When dealing with constraints, it's basically ksat
-//  - we have a list of requirements that are ANDed together
-//  - each is a list of predicates that are ORed together
-//  - each predicate has a continuation for when it's picked
-//  - need a way to thin the crowd, efficiently check compat, etc
-// Predicate is a four-element array [pat0,pat1,pal2,pal3]
-// Rather than a continuation we could go through all the slots again
-
-// class Constraints {
-//   constructor() {
-//     // Array of pattern table options.  Null indicates that it can be anything.
-//     //
-//     this.patterns = [[null, null]];
-//     this.palettes = [[null, null]];
-//     this.flyers = 0;
-//   }
-
-//   requireTreasureChest() {
-//     this.requireOrderedSlot(0, TREASURE_CHEST_BANKS);
-//   }
-
-//   requireOrderedSlot(slot, set) {
-
-//     if (!this.ordered) {
-
-//     }
-// // TODO
-//     this.pat0 = intersect(this.pat0, set);
-
-//   }
-
-// }
-
-// const intersect = (left, right) => {
-//   if (!right) throw new Error('right must be nontrivial');
-//   if (!left) return right;
-//   const out = new Set();
-//   for (const x of left) {
-//     if (right.has(x)) out.add(x);
-//   }
-//   return out;
-// }
-
-
-// useful for debug even if not currently used
-const [] = [hex];
+import { Assembler } from './asm/assembler';

```

Comment:
This whole file is marked as diff'd for some reason - did you accidentally change the line endings to DOS style?  Please change it back so as to avoid unnecessary merge conflicts.